Client/Provider/InstagramClient.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace KnpU\OAuth2ClientBundle\Client\Provider;
+
+use KnpU\OAuth2ClientBundle\Client\OAuth2Client;
+use League\OAuth2\Client\Provider\InstagramResourceOwner;
+use League\OAuth2\Client\Provider\ResourceOwnerInterface;
+use League\OAuth2\Client\Token\AccessToken;
+
+class InstagramClient extends OAuth2Client
+{
+    /**
+     * @param AccessToken $accessToken
+     * @return InstagramResourceOwner
+     */
+    public function fetchUserFromToken(AccessToken $accessToken)
+    {
+        return parent::fetchUserFromToken($accessToken);
+    }
+
+    /**
+     * @return InstagramResourceOwner|ResourceOwnerInterface
```

Comment:
You can remove the `| ResourceOwnerInterface`, since InstagramResourceOwner implements this